youtube_dl/extractor/kanal2.py

```diff
@@ -0,0 +1,122 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from datetime import datetime
+import re
+import time
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class Kanal2IE(InfoExtractor):
+    SUBTITLE_DATE_RE = re.compile(r'\((\d{2}\.\d{2}\.\d{4}\s\d{2}:\d{2})\)$')
+
+    _VALID_URL = r'(?P<base>https?://.+\.postimees\.ee)[a-zA-Z0-9/._-]+\?[a-zA-Z0-9=&._-]*id=(?P<id>[a-zA-Z0-9_-]+)[^ ]*'
+    _TESTS = [
+        {
+            'note': 'Test standard url (#18547)',
+            'url': 'https://kanal2.postimees.ee/pluss/video/?id=40792',
+            'md5': 'cecaf3e17706d725b1f23e886b67f8d3',
+            'info_dict': {
+                'id': '40792',
+                'ext': 'mp4',
+                'title': 'Aedniku aabits / Osa 53  (05.08.2016 20:00)',
+                'thumbnail': 'https://kanal-dl.babahhcdn.com/kanal/2016/08/05/0053_HNqKsIA/img/2.jpg',
+                'description': 'Aedniku aabits" on saade, mis pakub kaasaelamist ja teadmisi nii algajatele, kui juba kogenud rohenäppudele. Kõik alates vajalikest näpunäidetest, nutikatest lahendustest, uudistoodetest kuni taimede hingeeluni ning aias kasutatava tehnikani välja.',
+                'upload_date': '20160805',
+                'timestamp': 1470416400,
+            }
+        },
+        {
+            'note': 'Test preview url (#18547)',
+            'url': 'http://kanal2.postimees.ee/pluss/preview?id=40744',
+            'md5': 'e1dcc6e39d17a3f04749a8158db26377',
+            'info_dict': {
+                'id': '40744',
+                'ext': 'mp4',
+                'title': 'Kaunis Dila / Osa 50  (10.08.2016 19:00)',
+                'thumbnail': 'https://kanal-dl.babahhcdn.com/kanal/2018/12/05/16_300_00208_0050-Kaunis_Dila_hamdY9I/img/2.jpg',
+                'description': 'Riza ei tea, mis oht teda ja ta pere Selcuki näol varitseb. Azer kahtlustab, et Fatma elus on uus mees ja on valmis kõigeks, et ta endale tagasi võita. See tekitab aga Arzus suurt hirmu.',
+                'timestamp': 1470844800,
+                'upload_date': '20160810',
+            }
+        },
+    ]
+
+    def _real_extract(self, url_):
+        def get_title(info):
+            title = info['title']
+
+            if info['subtitle']:
+                title += ' / ' + info['subtitle']
+
+            return title
+
+        def get_timestamp(subtitle):
+            # Extract timestamp from:
+            #  "subtitle": "Osa 53  (05.08.2016 20:00)",
+            match = self._search_regex(self.SUBTITLE_DATE_RE, subtitle, 'dateandtime', default=None)
+            if not match:
+                return None
+
+            # https://stackoverflow.com/a/27914405/2314626
+            date = datetime.strptime(match, '%d.%m.%Y %H:%M')
+            unixtime = time.mktime(date.timetuple())
```

Comment:
how do I pass timezone? it needs to be local time (`Europe/Tallinn`)

Input = `05.08.2016 20:00`

```
AssertionError: Invalid value for field timestamp, expected 1470416400, got 1470427200
```

```
$ utime 1470427200
Fri Aug  5 23:00:00 2016 +0300
$ utime 1470416400
Fri Aug  5 20:00:00 2016 +0300
```

Reply:
tz data to input string: the injection must be in middle of the string if looking at `extract_timezone` implementation. this goes into implementation details. and the extract_timezone does not take symbolic names.

adjusting delta needs to be aware of DST. 

and the types are not compatible:
```python
        timestamp = unified_timestamp(match, day_first=True)
        timestamp += datetime.timedelta()
```

> `TypeError: unsupported operand type(s) for +=: 'int' and 'datetime.timedelta'`

- timestamp = int
- timedelta = datetime.timedelta

Reply:
seems to me that only way to get this done **correctly** regardless of your local timezone, is to use [pytz](http://pytz.sourceforge.net/) module. but this would be new dependency in this project. and timezones need to be updated regularily to catch up political changes, so bundling is not an option. optional dependency to get it almost right is okay?

```python
>>> import pytz
>>> import datetime
>>> tz = pytz.timezone('Europe/Tallinn')
>>> dt = datetime.datetime(2016, 8, 5, 20, 0, 0)
>>> dt
datetime.datetime(2016, 8, 5, 20, 0)
>>> offset = tz.utcoffset(dt)
>>> offset
datetime.timedelta(0, 10800)
>>>
```

but then need to convert timestamp back to something that `datetime.timedelta` is able to manipulate. perhaps `timestamp += delta.microseconds`

Reply:
and why should i use the `unified_timestamp` with all that fuzzy matching, if my format is known and does not need such heuristic searches (might even mismatch and make things worse). it doesn't even have `format` or `timezone` `kwargs` to have benefit of code de-duplication.

Reply:
tried with this for now: 9c3ac60fc